### PR TITLE
Vector homework done, waiting for review

### PR DESCRIPTION
**Pull request by Edgar6455**

### CODE REVIEW

File: Vector_Class/Vector.h [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+            arr2[i] = arr[i];
+
+        delete[] arr;
+        arr = arr2;
+    }
+
+
+public:
+
+    explicit Vector(size_t size = 0, T elem = 0) : size_(size), capacity_(size_)
```
Comment (Garnik645) [inline Vector_Class/Vector.h]: do not assign 0 to elem, elem might not a be a number

File: Vector_Class/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.10)
+project(Vector_Class)
+
+set(CMAKE_CXX_STANDARD 17)
+
+add_executable(Vector_Class main.cpp
+        Vector.h
```
Comment (Garnik645) [inline Vector_Class/CMakeLists.txt:7]: no need for Vector.h in cmake executable list

File: Vector_Class/Vector.h [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+    {
+        size_ = 0;
+        capacity_ = 0;
+        delete[] arr;
+    }
+
+
+    Vector(const Vector<T>& other)
+    {
+        delete[] arr;
```
Comment (Garnik645) [inline Vector_Class/Vector.h]: no need for delete, when constructor is called, arr is **nullptr**

File: Vector_Class/Vector.h [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        capacity_ = other.capacity_;
+
+        for (int i = 0; i < size_; ++i)
+            arr[i] = other.arr[i];
+
+    }
+
+    Vector(Vector<T>&& other)
+    {
+        delete[] arr;
```
Comment (Garnik645) [inline Vector_Class/Vector.h]: same here

File: Vector_Class/Vector.h [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+        other.arr = nullptr;
+    }
+
+    Vector<T> &operator=(const Vector<T> &other)
+    {
+        if (this == &other)
+            return *this;
+
+        delete[] arr;
+        arr = new T[capacity_];
```
Comment (Garnik645) [inline Vector_Class/Vector.h]: new array should be created with other.capacity, that is
```
arr = new T[other.capacity];
```

File: Vector_Class/Vector.h [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        capacity_ = other.capacity_;
+        arr = other.arr;
+        other.arr = nullptr;
+
+        return *this;
+    }
+
+    T &operator[](size_t index)
+    {
+        if (index >= size_) {
```
Comment (Garnik645) [inline Vector_Class/Vector.h]: operator[] does not check boundaries, only **at** does

File: Vector_Class/Vector.h [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+        if (index >= size_)
+        {
+            std::cerr << "Array index out of bound, exiting.";
+            exit(0);
+        }
+
+        return arr[index];
+    }
+
+    void push_back(const T& elem)
```
Comment (Garnik645) [inline Vector_Class/Vector.h:133]:
```c++
void push_back(T&& elem)
```
is missing

File: Vector_Class/Vector.h (hunk 1 above)
Comment (Edgar6455) [reply Vector_Class/Vector.h]: 👍

File: Vector_Class/Vector.h (hunk 3 above)
Comment (Edgar6455) [reply Vector_Class/Vector.h]: 👍

File: Vector_Class/Vector.h (hunk 4 above)
Comment (Edgar6455) [reply Vector_Class/Vector.h]: 👍

File: Vector_Class/Vector.h (hunk 6 above)
Comment (Edgar6455) [reply Vector_Class/Vector.h]: 👍

File: Vector_Class/Vector.h (hunk 5 above)
Comment (Edgar6455) [reply Vector_Class/Vector.h]: 👍

**Pull request merged**